### PR TITLE
Map and document the Box class

### PR DESCRIPTION
**Pull request by enbrain**

### CODE REVIEW

File: mappings/net/minecraft/util/math/Box.mapping [hunk 1]
```diff
@@ -76,33 +94,71 @@ CLASS net/minecraft/class_238 net/minecraft/util/math/Box
...
+		ARG 1 maxX
+	METHOD method_35578 withMaxY (D)Lnet/minecraft/class_238;
+		COMMENT Creates a box with the maximum Y provided and all other coordinates
+		COMMENT of this box.
+		ARG 1 maxY
+	METHOD method_35579 withMaxZ (D)Lnet/minecraft/class_238;
+		COMMENT Creates a box with the maximum Z provided and all other coordinates
+		COMMENT of this box.
+		ARG 1 maxZ
+	METHOD method_35580 (DDD)Lnet/minecraft/class_238;
```
Comment (liach) [inline mappings/net/minecraft/util/math/Box.mapping]: Should name this one. This one is the opposite of expand, like how shrink is the opposite of stretch.

Stretch/shrink only affects one side (min side if input <0, max side if input >0), while expand/this one (maybe name this `contract`? dunno) affects both min and max.

File: mappings/net/minecraft/util/math/Box.mapping [hunk 2]
```diff
@@ -76,33 +94,71 @@ CLASS net/minecraft/class_238 net/minecraft/util/math/Box
...
 		ARG 1 box
 	METHOD method_995 getAverageSideLength ()D
 	METHOD method_996 offset (Lnet/minecraft/class_2338;)Lnet/minecraft/class_238;
+		COMMENT Creates a box that is translated by {@code blockPos.getX()}, {@code
+		COMMENT blockPos.getY()}, {@code blockPos.getZ()} on each axis from this box.
 		ARG 1 blockPos
 	METHOD method_997 offset (Lnet/minecraft/class_243;)Lnet/minecraft/class_238;
-		ARG 1 vec3d
+		COMMENT Creates a box that is translated by {@code vec.x}, {@code vec.y}, {@code
+		COMMENT vec.z} on each axis from this box.
```
Comment (liach) [inline mappings/net/minecraft/util/math/Box.mapping:166]: I suggest javadoc linking to `@see #offset(double, double, double)`

File: mappings/net/minecraft/util/math/Box.mapping [hunk 3]
```diff
@@ -76,33 +94,71 @@ CLASS net/minecraft/class_238 net/minecraft/util/math/Box
...
+		COMMENT corners.
+		ARG 1 pos1
+		ARG 2 pos2
 	METHOD method_994 intersects (Lnet/minecraft/class_238;)Z
+		COMMENT Checks if this box intersects the given box.
 		ARG 1 box
 	METHOD method_995 getAverageSideLength ()D
 	METHOD method_996 offset (Lnet/minecraft/class_2338;)Lnet/minecraft/class_238;
+		COMMENT Creates a box that is translated by {@code blockPos.getX()}, {@code
+		COMMENT blockPos.getY()}, {@code blockPos.getZ()} on each axis from this box.
```
Comment (liach) [inline mappings/net/minecraft/util/math/Box.mapping:160]: Same javadoc `@see #offset(double, double, double)` link suggestion

File: mappings/net/minecraft/util/math/Box.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/util/math/Box.mapping]: I also prefer adding javadoc link between expand/contract and shrink/stretch to indicate their relations.

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/util/math/Box.mapping [hunk 4]
```diff
@@ -76,33 +97,76 @@ CLASS net/minecraft/class_238 net/minecraft/util/math/Box
...
 	METHOD method_991 union (Lnet/minecraft/class_238;)Lnet/minecraft/class_238;
+		COMMENT Creates the minimum box that contains this box and the given box.
 		ARG 1 box
 	METHOD method_992 raycast (Lnet/minecraft/class_243;Lnet/minecraft/class_243;)Ljava/util/Optional;
 		ARG 1 min
 		ARG 2 max
 	METHOD method_993 intersects (Lnet/minecraft/class_243;Lnet/minecraft/class_243;)Z
-		ARG 1 from
-		ARG 2 to
+		COMMENT Checks if this box intersects the box of of the given positions as
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/Box.mapping]:
```suggestion
		COMMENT Checks if this box intersects the box of the given positions as
```

**Review by YanisBft — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Review by haykam821 — APPROVED** [review]

**Pull request merged**